os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -250,95 +251,105 @@ static void up_dumpstate(void)
 {
 	struct tcb_s *rtcb = this_task();
 	uint32_t sp = up_getsp();
-	uint32_t ustackbase;
-	uint32_t ustacksize;
-#if CONFIG_ARCH_INTERRUPTSTACK > 3
-	uint32_t istackbase;
-	uint32_t istacksize;
-#endif
+	uint32_t stackbase = 0;
+	uint32_t stacksize = 0;
+	uint32_t istackbase = 0;
+	uint32_t istacksize = 0;
+	uint8_t irqNum;
 
-	/* Get the limits on the user stack memory */
-
-	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
-	ustacksize = (uint32_t)rtcb->adj_stack_size;
+	/* Get the limits for each type of stack */
 
+	stackbase = (uint32_t)rtcb->adj_stack_ptr;
+	stacksize = (uint32_t)rtcb->adj_stack_size;
 #if CONFIG_ARCH_INTERRUPTSTACK > 3
-	/* Get the limits on the interrupt stack memory */
-
 	istackbase = (uint32_t)&g_intstackbase;
 	istacksize = (CONFIG_ARCH_INTERRUPTSTACK & ~3);
-
-	/* Show interrupt stack info */
-
-	lldbg("sp:     %08x\n", sp);
-	lldbg("IRQ stack:\n");
-	lldbg("  base: %08x\n", istackbase);
-	lldbg("  size: %08x\n", istacksize);
-#ifdef CONFIG_STACK_COLORATION
-	lldbg("  used: %08x\n", up_check_intstack());
 #endif
+	bool isIrqAssert = false;
+	bool isSpCorrupt = false;
 
-	/* Does the current stack pointer lie within the interrupt
-	 * stack?
+	/* Check if the assert location is in user thread or IRQ handler.
+	 * If the irq_num is lesser than NVIC_IRQ_USAGEFAULT, then it is
+	 * a fault and not an irq.
 	 */
-
-	if (sp <= istackbase && sp > istackbase - istacksize) {
-		/* Yes.. dump the interrupt stack */
-
-		up_stackdump(sp, istackbase);
+	if (g_irq_nums[2] || (g_irq_nums[1] && g_irq_nums[0] > NVIC_IRQ_USAGEFAULT)) {
+		/* Assert in nested irq */
+		irqNum = 1;
```

Comment:
Need to check more conditions to determine the irqNum. 
Please split this `if` statement as below:

```
if (g_irq_nums[2]) {
    //Nested case
    if (g_irq_nums[0] > NVIC_IRQ_USAGEFAULT)
        irqNum = 0;
    else 
        irqNum = 1;
} else if (g_irq_nums[1] && g_irq_nums[0] > NVIC_IRQ_USAGEFAULT) {
    //Nested case
    irqNum = 0;
}
```